django/utils/timesince.py

```diff
@@ -24,20 +24,25 @@
 )
 
 
-def timesince(d, now=None, reversed=False, time_strings=None):
+def timesince(d, now=None, reversed=False, time_strings=None, depth=2):
     """
-    Take two datetime objects and return the time between d and now as a nicely
-    formatted string, e.g. "10 minutes". If d occurs after now, return
-    "0 minutes".
+    Take two datetime objects and return the time between d and now as a formatted string.
 
+    If d occurs after now, return "0 minutes".
     Units used are years, months, weeks, days, hours, and minutes.
-    Seconds and microseconds are ignored.  Up to two adjacent units will be
-    displayed.  For example, "2 weeks, 3 days" and "1 year, 3 months" are
-    possible outputs, but "2 weeks, 3 hours" and "1 year, 5 days" are not.
+    Seconds and microseconds are ignored.
 
-    `time_strings` is an optional dict of strings to replace the default
-    TIME_STRINGS dict.
+    Only adjacent units will be displayed.
+        Valid   : "2 weeks, 3 days" and "1 year, 3 months"
+        Invalid : "2 weeks, 3 hours" and "1 year, 5 days"
 
+    `depth` is an optimal integer to change the number of adjacent units to display.
+        depth = 1 returns "2 weeks"
+        depth = 2 returns "2 weeks, 3 days"
+        depth = 3 returns "2 weeks, 3 days, 3 hours"
+        depth = 4 returns "2 weeks, 3 days, 3 hours, 13 minutes"
+
+    `time_strings` is an optional dict of strings to replace the default TIME_STRINGS dict.
```

Comment:
Please revert change in wrapping, and wrap new docstring and comments at 79 chars.